libvast/include/vast/cast.hpp

```diff
@@ -243,6 +257,647 @@ struct cast_helper<record_type, record_type> {
+    -> caf::expected<void> {
+    return {};
+  }
+
+  static auto cast_value(const bool_type&, bool value,
+                         const double_type&) noexcept -> caf::expected<double> {
+    return static_cast<double>(value);
+  }
+
+  static auto
+  cast(const bool_type&, std::shared_ptr<type_to_arrow_array_t<bool_type>>,
+       const double_type&) noexcept
+    -> std::shared_ptr<type_to_arrow_array_t<double_type>> {
+    die("unimplemented");
+  }
+};
+
+template <>
+struct cast_helper<double_type, bool_type> {
+  static auto can_cast(const double_type&, const bool_type&) noexcept
+    -> caf::expected<void> {
+    return {};
+  }
+
+  static auto cast_value(const double_type&, double value,
+                         const bool_type&) noexcept -> caf::expected<bool> {
+    if (value != 0.0 and value != 1.0)
+      return caf::make_error(ec::convert_error,
+                             fmt::format("unable to convert {} into a double. "
+                                         "Only '0.0' and '1.0' are supported.",
+                                         value));
+    return static_cast<bool>(value);
+  }
+
+  static auto
+  cast(const double_type&, std::shared_ptr<type_to_arrow_array_t<double_type>>,
+       const bool_type&) noexcept
+    -> std::shared_ptr<type_to_arrow_array_t<bool_type>> {
+    die("unimplemented");
+  }
+};
+
+template <>
+struct cast_helper<int64_type, double_type> {
+  static auto can_cast(const int64_type&, const double_type&) noexcept
+    -> caf::expected<void> {
+    return {};
+  }
+
+  static auto cast_value(const int64_type&, int64_t value,
+                         const double_type&) noexcept -> caf::expected<double> {
+    constexpr auto max_loseless_integer
+      = (int64_t{1} << std::numeric_limits<double>::digits) - 1;
+    if (value > max_loseless_integer or value < -(max_loseless_integer)) {
+      return caf::make_error(ec::convert_error,
+                             "unable to convert {} into a double. The "
+                             "conversion may result in loss of precision");
+    }
+    return static_cast<double>(value);
+  }
+
+  static auto
+  cast(const int64_type&, std::shared_ptr<type_to_arrow_array_t<int64_type>>,
+       const double_type&) noexcept
+    -> std::shared_ptr<type_to_arrow_array_t<double_type>> {
+    die("unimplemented");
+  }
+};
+
+template <>
+struct cast_helper<double_type, int64_type> {
+  static auto can_cast(const double_type&, const int64_type&) noexcept
+    -> caf::expected<void> {
+    return {};
+  }
+
+  static auto cast_value(const double_type&, double value,
+                         const int64_type&) noexcept -> caf::expected<int64_t> {
+    if (std::isinf(value))
+      return caf::make_error(ec::convert_error,
+                             fmt::format("unable to convert {} into int64. The "
+                                         "input must be a finite number",
+                                         value));
```

Comment:
Generally for error messages, prefer the format `message: reason`. For this concrete example, I'd rewrite it to:

```suggestion
                             fmt::format("unable to convert {} into int64: the "
                                         "input must be a finite number",
                                         value));
```

Can you apply this throughout the file?

Reply:
Of course!